06_neighborhoods/Transportation/transit_trips_county.R

```diff
@@ -0,0 +1,203 @@
+# look at summary stats
+summary(transit_trips_county_2015$transit_trips_80ami)
+# examine outliers
+transit_trips_county_2015_outliers <- transit_trips_county_2015 %>% 
+  filter(transit_trips_80ami>250) 
+# 1057 trips in 38059 county -- Morton County, North Dakota -- seems to be a railroad town...
+# 1300 trips in 36047 -- Kings County, New York -- this is Brooklyn, makes total sense
+# all the highest ones(1000+) are in New York counties - checks out
+# 313 trips in 06037 -- Los Angeles county, California - checks out
+
+# Use stopifnot to check if all values in "transit_trips_county_2015" are non-negative
+stopifnot(min(transit_trips_county_2015$transit_trips_80ami, na.rm = TRUE) >= 0)
+# Good to go
+
+# Find indices of missing values for the "transit_trips_80ami" variable
+missing_indices <- which(is.na(transit_trips_county_2015$transit_trips_80ami))
+# Print observations with missing values
+if (length(missing_indices) > 0) {
+  cat("Observations with missing values for transit_trips_80ami:\n")
+  print(transit_trips_county_2015[missing_indices, , drop = FALSE])
+} else {
+  cat("No missing values for transit_trips_county_2015.\n")
+}
+# One missing value: Loving County, Texas (FIPS 48301)
+
+
+# County-Level Transit Trips 2019
+ggplot(transit_trips_county_2019, aes(x=transit_trips_80ami)) + geom_histogram(binwidth=5) + labs(y="number of counties", x="Annual Transit Trips for the Regional Moderate Income Household, 2019")
+# looks similar, checks out
+# look at summary stats
+summary(transit_trips_county_2019$transit_trips_80ami)  
+# examine outliers
+transit_trips_county_2019_outliers <- transit_trips_county_2019 %>% 
+  filter(transit_trips_80ami>250) 
+# 1160 trips in 42101 county - Bowling Green, Kentucky. GObg, Bowling Green's public transit system, provides para-transit service throughout the City of Bowling Green
+# 2105 trips in 25025 county - Suffolk County, MA - Boston metro area, makes sense
+# 1150 trips in 11001 county - Washington DC, makes sense
+# all the highest ones(1000+) are in New York counties - checks out
+
+# Use stopifnot to check if all values in "transit_trips_county_2019" are non-negative
+stopifnot(min(transit_trips_county_2019$transit_trips_80ami, na.rm = TRUE) >= 0)
+# Good to go
+
+# Find indices of missing values for the "transit_trips_80ami" variable
+missing_indices <- which(is.na(transit_trips_county_2019$transit_trips_80ami))
+# Print observations with missing values
+if (length(missing_indices) > 0) {
+  cat("Observations with missing values for transit_trips_80ami:\n")
+  print(transit_trips_county_2019[missing_indices, , drop = FALSE])
+} else {
+  cat("No missing values for transit_trips_county_2019\n")
+}
+# No missing values for 2019
+
+#####################################################################
+
+# Create national percentile ranking for 'transit_trips_80ami'
+transit_trips_county_2015$rank <- rank(transit_trips_county_2015$transit_trips_80ami)
```

Comment:
This code is a mix of base R and tidyverse syntax. It's good practice to stick with one. Since you already have some tidyverse I would suggest sticking with that. This section of code can be re-written more efficient and even more so if it's wrapped in a function. 

```
transit_trips_county_2015 <- transit_trips_county_2015 %>%
  mutate(rank = rank(transit_trips_80ami),
         percentile_rank = (rank - 1) / (n() - 1) * 100,
         percentile_rank = round(percentile_rank, 2)) %>%
  rename(index_transit_trips = percentile_rank)
```